### DISCUSSION TITLE
Why do shapes not extend Animatable()?

### DISCUSSION THREAD

**Discussion by Omixxx** [openrndr/openrndr#408] [Q&A] [answered]
May be a stupid question, I'm pretty new to OpenRndr and software development (and kotlin 😅), but why aren't the attributes of shape classes (rectangle, circle, line, ect) animable or settable outside of the constructor? i mean, doing something like this I guess would be cool.

```kotlin
    oliveProgram {
        val rect = Rectangle(drawer.bounds.center, 10.0)
        
        extend {
            drawer.rectangle(rect)
            rect.width.animte(rect.width.times(2), 5000)
        }
    }
```

As for now, as far as what I understand, I need to create a wrapper for the properties that I want to animate, and then use it to create an animation.

```kotlin
    oliveProgram {
        class AnimatedRectangle(var width: Double) : Animatable() {
            fun grow() {
                ::width.animate(width.times(2.0), 5000)
            }
        }

        val rectAnimation = AnimatedRectangle(10.0)
        extend {
            rectAnimation.updateAnimation()
            drawer.rectangle(drawer.bounds.center, rectAnimation.width)
            rectAnimation.grow()
        }
    }
```

Am I missing something about the philosophy of "how things should be done" in openRndr?

Thanks for your help 

(ﾉ◕ヮ◕)ﾉ*:･ﾟ✧

**Comment by hamoid** [comment, marked as answer]
Hi! :-)

I think it may be related to two ideas:

1. A `Rectangle` instance is a mathematical Rectangle. Is not something necessarily visible, but something with a position, a width and a height. Drawing rectangles, animating rectangles and doing rectangle calculations are 3 separate topics, and a program may involve one of these areas but not the others. That's why it wouldn't make so much sense to make them animatable. `Rectangle` has various useful methods, for instance, to know if a point is inside or not. `Drawer` knows how to draw certain entities like `Rectangle` and `Circle`.
2. That brings to another idea: inmutability. Why are most instances in OPENRNDR inmutable? That's a design decision of its designers. Inmutability can avoid certain kinds of problems and make programs simpler to debug. You can read about inmutability online.

Feel free to ask more about these or other topics! :)

**Reply by Omixxx** [reply]
Thank you for your answer! It makes more sense now 🙏
